packages/terra-button/src/Button.jsx

```diff
@@ -115,225 +115,197 @@ const defaultProps = {
   variant: ButtonVariants.NEUTRAL,
 };
 
-class Button extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { active: false, focused: false };
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-    this.handleKeyUp = this.handleKeyUp.bind(this);
-    this.handleOnBlur = this.handleOnBlur.bind(this);
-    this.handleFocus = this.handleFocus.bind(this);
-    this.handleMouseDown = this.handleMouseDown.bind(this);
-    this.handleClick = this.handleClick.bind(this);
-
-    this.shouldShowFocus = true;
-  }
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    if (prevState.focused && nextProps.isDisabled) {
-      return { focused: false };
+const Button = ({
+  icon,
+  isBlock,
+  isCompact,
+  isDisabled,
+  isIconOnly,
+  isReversed,
+  text,
+  type,
+  variant,
+  href,
+  onClick,
+  onMouseDown,
+  onBlur,
+  onFocus,
+  onKeyDown,
+  onKeyUp,
+  refCallback,
+  title,
+  ...customProps
+}) => {
+  const theme = useContext(ThemeContext);
+
+  // We need to manually control active state for FF compatibility
+  const [isActive, setIsActive] = useState(false);
```

Comment:
Does this logic have any affect in other browsers? Would it make sense to only execute this logic if the browser is FireFox?

Reply:
Just checked and I think it only affects Firefox like the original comment.